messaging/src/main/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessor.java

```diff
@@ -132,7 +133,7 @@ protected PooledStreamingEventProcessor(PooledStreamingEventProcessor.Builder bu
         this.workerExecutor = builder.workerExecutorBuilder.apply(name);
         this.initialToken = builder.initialToken;
         this.tokenClaimInterval = builder.tokenClaimInterval;
-        this.maxClaimedSegments = builder.maxClaimedSegments;
+        this.maxSegmentProvider = builder.maxSegmentProvider!=null?builder.maxSegmentProvider: p -> Short.MAX_VALUE;
```

Comment:
I would prefer to have this as default in the builder, so we can just take the value from the builder at this point.

Reply:
Resolved. Moved the default to the builder.